judges/quality-of-service/quality-of-service.rb

```diff
@@ -32,15 +32,26 @@
   total = 0
   success = 0
   duration = 0
+  mttr = {}
   Fbe.unmask_repos.each do |repo|
     Fbe.octo.repository_workflow_runs(repo, created: ">#{f.since.utc.iso8601[0..9]}")[:workflow_runs].each do |json|
+      workflow_id = json[:workflow_id]
+      if json[:conclusion] == 'failure'
+        mttr[workflow_id] = { last_failed_at: json[:updated_at] }
```

Comment:
@Suban05 Why you use `updated_at`? This is exactly the end time of the completion workflow run?

Reply:
```
{:id=>10564816841,
 :name=>"zerocracy",
 :node_id=>"WFR_kwLOL6J6Ss8AAAACdbZPyQ",
 :head_branch=>"master",
 :head_sha=>"42f96b41d866b73351ee87fdac920ac252cf2e00",
 :path=>".github/workflows/zerocracy.yml",
 :display_title=>"zerocracy",
 :run_number=>2798,
 :event=>"schedule",
 :status=>"completed",
 :conclusion=>"success",
 :workflow_id=>97829964,
 :check_suite_id=>27615969050,
 :check_suite_node_id=>"CS_kwDOL6J6Ss8AAAAGbgo_Gg",
 :url=>
  "https://api.github.com/repos/zerocracy/judges-action/actions/runs/10564816841",
 :html_url=>
  "https://github.com/zerocracy/judges-action/actions/runs/10564816841",
 :pull_requests=>[],
 :created_at=>2024-08-26 18:04:49 UTC,
 :updated_at=>2024-08-26 18:27:02 UTC,
:actor=>
  {:login=>"yegor256",
   :id=>526301,},
 :run_attempt=>1,
 :referenced_workflows=>[],
 :run_started_at=>2024-08-26 18:04:49 UTC,
```

Maybe we need use `run_started_at` field ? @yegor256 What do you think?

Reply:
@Yegorov since there is no explicit field indicating the end time, I think `updated_at` indicates when the workflow has been completed. The `run_started_at` field indicates the launch date. We need the end date.

Reply:
@Yegorov I don't see where we can get the `duration` field...
what do you think about if we will check the `status` field and then can use `updated_at`? For example, if the status is completed then `updated_at` is end time

Reply:
@Suban05 duration can be obtained for example:
```ruby
Fbe.octo.workflow_run_usage('zerocracy/judges-action', 10581095703)
```

Reply:
@Yegorov I got you, thanks!